src/rust/ide/src/lib.rs

```diff
@@ -152,3 +152,42 @@ pub fn run_ide() {
         project_view.forget();
     });
 }
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use enso_protocol::project_manager::API;
+    use enso_protocol::project_manager::Client;
+
+    use wasm_bindgen_test::wasm_bindgen_test_configure;
+
+    wasm_bindgen_test_configure!(run_in_browser);
+
+
+    //#[wasm_bindgen_test::wasm_bindgen_test(async)]
+    #[allow(dead_code)]
+    async fn project_life_cycle() {
+        let ws        = WebSocket::new_opened("ws://localhost:30535").await;
+        let ws        = ws.expect("Couldn't connect to WebSocket server.");
+        let client    = Client::new(ws);
+        let _executor = setup_global_executor();
+
+        executor::global::spawn(client.runner());
+
+        let name     = "TestProject".to_string();
+        let creation = client.create_project(name).await.expect("Couldn't create project.");
+        let uuid     = creation.project_id;
+        let _address = client.open_project(uuid.clone()).await.expect("Couldn't open project.");
+        client.close_project(uuid.clone()).await.expect("Couldn't close project.");
+        client.delete_project(uuid).await.expect("Couldn't delete project.");
+        client.list_recent(10).await.expect("Couldn't list recent projects.");
+        // FIXME[dg]: project/listSample isn't implemented on the server-side yet.
+        //client.list_sample(10).await.expect("Couldn't list samples.");
+    }
+}
```

Comment:
Please, create a file in src/rust/ide/enso-protocol/tests and put this test there.

Reply:
@farmaazon to do that, I would need to move executor out of IDE.